src/components/upload/nz-basic-upload.component.ts

```diff
@@ -0,0 +1,149 @@
+import { Component, EventEmitter, Input, Output, OnInit, ElementRef, ViewChild } from '@angular/core';
+import { UidService } from './uid/uid.service';
+import defaultRequest from './request';
+import { attrAccept } from './utils';
+
+@Component({
+    selector: 'nz-basic-upload',
+    providers: [UidService],
+    templateUrl: './nz-basic-upload.component.html'
+})
+export class NzBasicUploadComponent implements OnInit {
+
+    _classMap;
+    _prefixCls = 'ant-upload';
+    _reqs = {};
+
+    @Input() accept: string;
+    @Input() action: string;
+    @Input() autoUpload = true;
+    @Input() beforeUpload: (file: File, files: FileList) => boolean | Promise<any> = null;
+    @Input() customRequest: any;
+    @Input() data: any;
+    @Input() disabled = false;
+    @Input() headers: string | { [name: string]: string | string[] } = {};
+    @Input() multiple = false;
+    @Input() name: string;
+    @Input() withCredentials = false;
+
+    @Output() onError = new EventEmitter<any>();
+    @Output() onProgress = new EventEmitter<any>();
+    @Output() onRemove = new EventEmitter<any>();
+    @Output() onStart = new EventEmitter<any>();
+    @Output() onSuccess = new EventEmitter<any>();
+
+    @ViewChild('input') inputElement: ElementRef;
+
+
+
+    constructor(private uidService: UidService) { }
+
+    ngOnInit() {
+        this.setClassMap();
+    }
+
+    setClassMap(): void {
+        this._classMap = {
+            [`${this._prefixCls}`]: true,
+            [`${this._prefixCls}-disabled`]: this.disabled,
+        };
+    }
+
+    onChange = (e) => {
+        const files: FileList = e.target.files;
+        if (!files) { return };
+        this.uploadFiles(files);
+    }
+
+    uploadFiles(files: FileList) {
+        const postFiles = Array.prototype.slice.call(files);
```

Comment:
there are no any magic on `Array.prototype.slice`. just use `files`

Reply:
@LinBoLen Thanks for your advice. `Array.prototype.slice` isn't useless.I try removing it,and get errors in the next line `foreach` function.